src/runtime/TemporalObject.h

```diff
@@ -0,0 +1,132 @@
+class TemporalCalendar : public Temporal {
+public:
+    explicit TemporalCalendar(ExecutionState& state);
+    explicit TemporalCalendar(ExecutionState& state, Object* proto);
+
+    bool isTemporalCalendarObject() const override
+    {
+        return true;
+    }
+
+    static TemporalCalendar* createTemporalCalendar(ExecutionState& state, String* id, Optional<Object*> newTarget = Optional<Object*>());
+    static bool isBuiltinCalendar(String* id);
+    static Value getBuiltinCalendar(ExecutionState& state, String* id);
+    static Value getISO8601Calendar(ExecutionState& state);
+    static Value toTemporalCalendar(ExecutionState& state, const Value& calendar);
+    static Value toTemporalCalendarWithISODefault(ExecutionState& state, const Value& calendar);
+    static Value parseTemporalCalendarString(ExecutionState& state, const Value& isoString);
+    static Value ISODaysInMonth(ExecutionState& state, const Value& year, const Value& month);
+    static bool isIsoLeapYear(ExecutionState& state, const Value& year);
+    static Value defaultMergeFields(ExecutionState& state, const Value& fields, const Value& additionalFields);
+    String* getIdentifier() const;
+    void setIdentifier(String* identifier);
+
+private:
+    String* identifier;
+};
+
+class TemporalPlainDate : public Temporal {
+public:
+    explicit TemporalPlainDate(ExecutionState& state, Value isoYear, Value isoMonth, Value isoDay, Optional<Value> calendarLike);
+    explicit TemporalPlainDate(ExecutionState& state, Object* proto, Value isoYear, Value isoMonth, Value isoDay, Optional<Value> calendarLike);
+
+    static Value createTemporalDate(ExecutionState& state, const Value& isoYear, const Value& isoMonth, const Value& isoDay, const Value& calendar, Optional<Object*> newTarget);
+    static bool isValidISODate(ExecutionState& state, const Value& year, const Value& month, const Value& day);
+
+    bool hasInitializedTemporalDate() const
+    {
+        return true;
+    }
+
+    int getYear() const { return y; }
+    int getMonth() const { return m; }
+    int getDay() const { return d; }
+
+private:
+    int y;
+    int m;
+    int d;
+    Value calendar;
+};
+
+class TemporalPlainTime : public Temporal {
+public:
+    explicit TemporalPlainTime(ExecutionState& state);
+    explicit TemporalPlainTime(ExecutionState& state, Object* proto);
+
+    static Value createTemporalTime(ExecutionState& state, size_t argc, Value* argv, Optional<Object*> newTarget);
+    static bool isValidTime(ExecutionState& state, const Value& hour, const Value& minute, const Value& second, const Value& millisecond, const Value& microsecond, const Value& nanosecond);
+
+    void setTime(int h, int m, int s, int ms, int qs, int ns);
+    void setCalendar(ExecutionState& state, String* calendar, Optional<Object*> newTarget);
+
+
+private:
+    int year;
+    int month;
+    int day;
+    int hour;
+    int minute;
+    int second;
+    int millisecond;
+    int microsecond;
+    int nanosecond;
+
```

Comment:
If you can use smaller type like char
You can use it
+
please add "m_" before member variable